src-built-in/components/toolbar/stores/toolbarStore.js

```diff
@@ -246,33 +245,169 @@ class _ToolbarStore {
-				self.Store.setValue({ field: "searchActive", value: true });
-			});
+		/**
+		 * hotkeyActions
+		 * A string can be provided by a hotkey and the corresponding function gets returned.
+		 * ?This is the only way to be able to send actions from JSON.
+		 * ?This flexibility allows us to change the action via config.
+		 * @param {string} action
+		 */
+		const hotkeyActions = action => {
+			const self = this;
+			switch (action) {
+
+				case 'SHOW_TOOLBAR':
+					return function showToolbar() {
+						self.Store.setValue({ field: "searchActive", value: false });
+						self.showToolbarAtFront();
+					}
+
+				case 'HIDE_TOOLBAR':
+					return self.hideToolbar
+
+				case 'BRING_WINDOWS_TO_FRONT':
+					return function bringWindowsToFront() {
+						FSBL.Clients.LauncherClient.bringWindowsToFront()
+					}
+
+				case 'MINIMIZE_ALL':
+					return function minimizeAll() {
+						FSBL.Clients.WorkspaceClient.minimizeAll()
+					}
+
+				case 'SHOW_SEARCH':
+					return function showSearch() {
+						self.showToolbarAtFront()
+						self.Store.setValue({ field: "searchActive", value: true });
+					}
+
+				default:
+					return null
+			}
+		}
+
+		/**
+		 * setHotkey
+		 *
+		 * This function does a some validation checks and then sets the global hotkey to the keyCombination with the action passed to it.
+		 * @param hotkey array of keys
+		 * @param action function to be assigned to the hotkey
+		 */
+		const setHotkey = ({ name, keyCombination, action, enabled }) => {
+			try {
+				const getAction = hotkeyActions(action)
+				// ensure the action is a function, the key combo is an array and not empty
+				if (typeof getAction === "function" && Array.isArray(keyCombination) && keyCombination.length && enabled) {
+					FSBL.Clients.HotkeyClient.addGlobalHotkey(keyCombination, getAction, () => { FSBL.Clients.Logger.log(`Hotkey added - ${name} ${keyCombination}`) })
+				} else {
+					throw 'The key combination or action supplied are not valid.'
+				}
+			} catch (err) {
+				FSBL.Clients.Logger.error(`Failed to add the hotkey ${keyCombination} for ${name}.`, err)
+			}
+		}
+
+		/**
+		 * updateDefaultHotkeysFromConfig
+		 *
+		 * This function takes the hotkeys from the config file and updates / merges the default hotkeys with it's values.
+		 * @param {Array} configHotkeys
+		 * @param {Array} defaultHotkeys
+		 */
+		const overrideDefaultHotkeysWithConfigHotkeys = (configHotkeys, defaultHotkeys) =>
+			defaultHotkeys.map(defaultHotkey => {
+
+				// This checks to see if there is an object inside the default hotkey array that matches
+				// one from the configHotkeys - validation check
+				const config = configHotkeys
+					.find(configHotkey => defaultHotkey.name === configHotkey.name)
+
+				// return one object by merging the two objects into one object.
+				// If there is a value already set in both objects then the right-hand side (configHotkeys)
+				// overwrites the left (hotkeys)
+				return { ...defaultHotkey, ...config }
```

Comment:
Wouldn't this make it impossible to remove a default hotkey?

Reply:
I don't think so as this is only just merging the objects (no hotkeys are set yet).
The hotkeys get set in a function lower down, does this sound correct? 
I have also just changed the block comment to make this a bit clearer.